pkg/odo/cli/component/push.go

```diff
@@ -88,8 +88,17 @@ func (po *PushOptions) GetComponentContext() string {
 // Complete completes push args
 func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 	po.CompleteDevfilePath()
+	devfileExists := util.CheckPathExists(po.DevfilePath)
+
+	if !devfileExists {
+		// Note: Remove this deprecation warning once the S2I cleanup is done, see https://github.com/openshift/odo/issues/4932.
+		log.Deprecate(
```

Comment:
When should one expect this log to show up? I didn't see it when I did `odo push` after `odo create --s2i nodejs`:
```sh

$ odo create nodejs --s2i
Flag --s2i has been deprecated, support for the S2I components has been deprecated, consider creating a devfile component instead.
 ⚠  S2I components Deprecated  consider creating a devfile component instead
Validation
 ✓  Validating component [9ms]
Conversion
 ✓  Successfully generated devfile.yaml and env.yaml for provided S2I component

Please use `odo push` command to create the component with source deployed

$ odo push

Validation
 ✓  Validating the devfile [56849ns]

Creating Services for component nodejs-node-todo-znbo
 ✓  Services are in sync with the cluster, no changes are required

Creating Kubernetes resources for component nodejs-node-todo-znbo
 ✓  Waiting for component to start [6s]
 ✓  Links are in sync with the cluster, no changes are required
 ✓  Waiting for component to start [1ms]

Applying URL changes
 ✓  URL http-8080: http://http-8080-app-myproject.apps-crc.testing/ created

Syncing to component nodejs-node-todo-znbo
 ✓  Checking files for pushing [4ms]
 ✓  Syncing files to the component [776ms]

Executing devfile commands for component nodejs-node-todo-znbo
 ✓  Executing s2i-assemble command "/opt/odo/bin/s2i-setup && /opt/odo/bin/assemble-and-restart" [23s]
 ✓  Executing s2i-run command "/opt/odo/bin/run" [1s]

Pushing devfile component "nodejs-node-todo-znbo"
 ✓  Changes successfully pushed to component
```

Reply:
That's because devfile.yaml exists for `odo create --s2i nodejs`. 
You can expect it to show up with `odo create --s2i --git` or `odo create --s2i --binary`.